pkg/providers/pricing/pricing.go

```diff
@@ -54,7 +54,8 @@ type Provider struct {
 	region  string
 	cm      *pretty.ChangeMonitor
 
-	mu                 sync.RWMutex
+	muOnDemand         sync.RWMutex
+	muSpot             sync.RWMutex
 	onDemandPrices     map[string]float64
 	spotPrices         map[string]zonal
```

Comment:
```suggestion
	muOnDemand         sync.RWMutex
	onDemandPrices     map[string]float64
	
	muSpot             sync.RWMutex
	spotPrices         map[string]zonal

```

nit: style, mutexes typically sit directly above the object that they are managing, consider breaking them up and adding spacing to logically separate them